### PR TITLE
Enhancement: Implement VersionConstraintNormalizer

### LINKED ISSUE #16: Normalize or operator
According to composer standards, the or operator should be ` || `.

It would be great to add this to normalize. Samples:

- `^1.2|^2.0` => `^1.2 || ^2.0`
- `^1.2||^2.0` => `^1.2 || ^2.0`

Only the ` || ` is documented and Seldaek said it should be the default way, `|` works here historically. See also: https://github.com/composer/composer/issues/5257

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] implements a `VersionConstraintNormalizer`

Fixes #16.

### CODE REVIEW

**Comment by soullivaneuh** [comment]
> For example, trimming between version constraints

Yeah!

> ensuring a single space between them

Looks good, but maybe we should find official examples on getcomposer.org

**Comment by soullivaneuh** [comment]
I can't find official example, but I always see something like `^2.0,<=2.0.4` without any space. So it should be ok.

File: README.md [hunk 1]
```diff
@@ -79,6 +80,19 @@ sections, the `PackageHashNormalizer` will sort the content of these sections.
...
+
+If `composer.json` contains any *or* version constraints (`|` or `||`) in the  
+
+* `conflict`
+* `provide`
+* `replaces`
+* `require`
+* `require-dev`
+
+sections, the `VersionOrConstraintNormalizer` will ensure that a `||` is
```
Comment (soullivaneuh) [inline README.md]: ` || ` (with spaces around)

**Review by soullivaneuh — CHANGES_REQUESTED** [review]

**Comment by localheinz** [comment]
@Soullivaneuh 

Would you like to take another look?

**Comment by localheinz** [comment]
@Soullivaneuh 

I incorporated all examples I could find at https://getcomposer.org/doc/articles/versions.md and think of, I think this is good to merge (once the build has passed).

**Comment by soullivaneuh** [comment]
Sorry for the delay, but it looks great!

**Comment by localheinz** [comment]
Glad you like it, @Soullivaneuh!

**Pull request merged**